### DISCUSSION TITLE
Esp32 wrover-e + 2'' waveshare st7789

### DISCUSSION THREAD

**Discussion by AnRich26** [Bodmer/TFT_eSPI#2976] [Q&A - Sprites]
I am using Esp32 Wrover-E 16mb Flash 8Mb Psram connected to 2'' 320x240 Waveshare st7789, i am able to use the screen to display images and text. However when using sprites set at full screen i only get about 240x240 pushed to the screen i have checked user setup and done all the wright changes to use with the screen exept sprite full screen seems not to be working. My code is basic and is as follows:

#include <SPI.h>
#include <TFT_eSPI.h>

TFT_eSPI tft = TFT_eSPI();
TFT_eSprite bak = TFT_eSprite(&tft);

void setup() {
  // Init Serial Port
  Serial.begin(9600);
  // Init Tft
  tft.init();
  tft.setRotation(1);
  tft.setSwapBytes(true);
  tft.fillScreen(TFT_BLACK);
  Serial.println("Screen Init Done!");
}

void loop() {
  bak.createSprite(TFT_WIDTH,TFT_HEIGHT);
  //bak.setRotation(1);
  bak.setSwapBytes(true);
  bak.fillScreen(TFT_RED);
  bak.pushSprite(0,0);
}

**Comment by Bodmer** [comment]
The ESP32 will not have enough RAM unless the board has PSRAM fitted and enabled.

**Reply by AnRich26** [reply]
My esp32 wrover-e has bove flash and psram, when power on and press reset can see it init's the psram. However after playing around with my code found my problem, bak.fillScreen(TFT_RED); i was trying to fill screen and not sprite changed it to bak.fillSprite(TFT_RED); and got it working full screan. However when using another sprite on that one pusshing the sprite to the back sprite i get lot of flickering, and in my code thears no delay. All in all works very well.

**Reply by AnRich26** [reply]
Am getting next error from arduino droid:

/storage/emulated/0/ArduinoDroid/userlibraries/TFT_eSPI/Processors/TFT_eSPI_ESP32.h:14:25: fatal error: hal/gpio_ll.h: No such file or directory

**Reply by AnRich26** [reply]
And if i comment that part out all seems to compile fine.
